atc/worker/client.go

```diff
@@ -1,16 +1,38 @@
 package worker
 
 import (
+	"code.cloudfoundry.org/garden"
 	"code.cloudfoundry.org/lager"
+	"context"
+	"fmt"
+	"github.com/concourse/concourse/atc"
 	"github.com/concourse/concourse/atc/db"
+	"io"
+	"path"
+	"strconv"
 )
 
+const taskProcessID = "task"
+const taskExitStatusPropertyName = "concourse:exit-status"
+
 //go:generate counterfeiter . Client
 
 type Client interface {
 	FindContainer(logger lager.Logger, teamID int, handle string) (Container, bool, error)
 	FindVolume(logger lager.Logger, teamID int, handle string) (Volume, bool, error)
 	CreateVolume(logger lager.Logger, vSpec VolumeSpec, wSpec WorkerSpec, volumeType db.VolumeType) (Volume, error)
+	RunTaskStep(
+		context.Context,
+		lager.Logger,
+		db.ContainerOwner,
+		ContainerSpec,
+		WorkerSpec,
+		ContainerPlacementStrategy,
+		db.ContainerMetadata,
+		ImageFetcherSpec,
+		TaskProcessSpec,
+		chan string,
+	) (int, []VolumeMount, error)
```

Comment:
Why is `Run` returning `VolumeMount`?

Reply:
Because we need to [registerOutputs](https://github.com/concourse/concourse/blob/392c00e931b5b81db6a8a01b926edaf6a2dfa087/atc/exec/task_step.go#L388-L433) in the TaskStep. Decoupling volumes/storage from Running a step is something we would need to look into, but in the absence of a separate Storage implementation, we would have to return volumes this way.

For example, we would likely register "outputs" beforehand with a specific storage implementation, and then de-register them if the task failed. That would allow us to not depend on VolumeMounts in the future.